src/config/colorpickereditor.cpp

```diff
@@ -0,0 +1,161 @@
+{
+    ConfigHandler config;
+    m_color = config.drawColor();
+
+    m_gLayout = new QGridLayout(this);
+
+    m_colorpicker = new ColorPickerWidget(this);
+    m_gLayout->addWidget(m_colorpicker, 0, 0);
+
+    m_colorWheel = new color_widgets::ColorWheel(this);
+    m_colorWheel->setColor(m_color);
+    const int size = GlobalValues::buttonBaseSize() * 3.5;
+    m_colorWheel->setMinimumSize(size, size);
+    m_gLayout->addWidget(m_colorWheel, 1, 0);
+
+    QVBoxLayout* m_vLocalLayout1 = new QVBoxLayout();
+    m_vLocalLayout1->addStretch();
+
+    m_spinboxLabel = new QLabel(tr("Select Preset:"), this);
+    m_vLocalLayout1->addWidget(m_spinboxLabel);
+
+    m_spinbox = new SpinBox(this);
+    connect(m_spinbox,
+            QOverload<int>::of(&QSpinBox::valueChanged),
+            m_colorpicker,
+            [=](int val) {
+                m_selectedIndex = val;
+                m_colorpicker->updateSelection(val);
+            });
+    m_spinbox->setToolTip(tr("Select preset using the spinbox"));
+    m_vLocalLayout1->addWidget(m_spinbox);
+
+    m_deletePresetButton = new QPushButton(tr("Delete"), this);
+    m_deletePresetButton->setToolTip(
+      tr("Press button to delete the selected preset"));
+    connect(m_deletePresetButton,
+            &QPushButton::pressed,
+            this,
+            &ColorPickerEditor::onDeletePreset);
+    m_vLocalLayout1->addWidget(m_deletePresetButton);
+
+    m_vLocalLayout1->addStretch();
+
+    m_gLayout->addLayout(m_vLocalLayout1, 0, 1);
+
+    QVBoxLayout* m_vLocalLayout2 = new QVBoxLayout();
+    m_vLocalLayout2->addStretch();
+
+    m_addPresetLabel = new QLabel(tr("Add Preset:"), this);
+    m_vLocalLayout2->addWidget(m_addPresetLabel);
+
+    m_colorInput = new QLineEdit(this);
+    m_colorInput->setText(m_color.name(QColor::HexRgb));
+    m_colorInput->setToolTip(
+      tr("Enter color manually or select it using the color-wheel"));
+    connect(m_colorWheel,
+            &color_widgets::ColorWheel::colorSelected,
+            this,
+            [=](QColor c) {
+                m_color = c;
+                m_colorInput->setText(m_color.name(QColor::HexRgb));
+            });
+    m_vLocalLayout2->addWidget(m_colorInput);
+
+    m_addPresetButton = new QPushButton(tr("Add"), this);
+    m_addPresetButton->setToolTip(tr("Press button to add preset"));
+    connect(m_addPresetButton,
+            &QPushButton::pressed,
+            this,
+            &ColorPickerEditor::onAddPreset);
+    m_vLocalLayout2->addWidget(m_addPresetButton);
+
+    m_vLocalLayout2->addStretch();
+
+    m_gLayout->addLayout(m_vLocalLayout2, 1, 1);
+}
+
+void ColorPickerEditor::addPreset()
+{
+    ConfigHandler config;
+    QVector<QColor> colors = config.userColors();
+
+    if (colors.contains(m_color))
+        return;
+
+    colors << m_color;
+
+    const int maxPresetsAllowed = 10;
+
+    if (colors.size() > maxPresetsAllowed) {
+        QMessageBox::critical(this, tr("Error"), tr("Unable to add preset."));
```

Comment:
Adjust this error to state why the preset cannot be added "IE exceeded maximum" or something.